build-tools/trim-analyzers/trim-analyzers.props

```diff
@@ -0,0 +1,53 @@
+<!-- Import this to enable trim warnings of all kinds -->
+<Project>
+  <PropertyGroup>
+    <!-- Sets assembly metadata, enable analyzers -->
+    <IsTrimmable>true</IsTrimmable>
+    <EnableSingleFileAnalyzer>true</EnableSingleFileAnalyzer>
+    <EnableAotAnalyzer>true</EnableAotAnalyzer>
+    <!-- In app projects, tells ILLink to emit warnings as errors -->
+    <ILLinkTreatWarningsAsErrors>true</ILLinkTreatWarningsAsErrors>
+    <!--
+      Trim warnings, codes listed here:
+      * https://github.com/dotnet/runtime/blob/7403a062960d092c73ce3f07d3ff323ffdf7de43/src/tools/illink/src/linker/Resources/Strings.resx
+      * https://github.com/dotnet/docs/tree/9cb45cf9cd34f3b7259a023c3d92a124a87090d5/docs/core/deploying/trimming/trim-warnings
+    -->
+    <WarningsAsErrors>
+      $(WarningsAsErrors);
+      IL2000;IL2001;IL2002;IL2003;IL2004;
```

Comment:
I found it didn't work in a class library, unless I added these.

Let me look at an MSBuild log to see why.

Reply:
I can't find anything in the ILLink analyzer that would look for `$(ILLinkTreatWarningsAsErrors)` and promote the warnings to errors.

I was looking here:

* https://github.com/search?q=repo%3Adotnet/runtime%20IsMSBuildPropertyValueTrue&type=code

Reply:
Analyzer doesn't do this, if you need this for the analyzer you do need to use the Roslyn's properties - so `WarnAsError` or `TreatWarningsAsErrors` (if you can enable it for all warnings). The `ILLinkTreatWarningsAsErrors` is only for `ILLink`.

Reply:
I think what we have here is OK for now. We could consider `$(TreatWarningsAsErrors)` as some point, but we have ~31 long-standing warnings from binding the Android API surface. There are a lot fewer now than there used to be, so it might be reasonable to fix these in .NET 9.